calcom/cal.com
# fix: possible to have same slug for team and user in an organization 9985 cal 2122

## What does this PR do?

- Moves invite token validation before user creation
- If token has teamId we collect every team.slug in the org or just use that team.slug so a username cannot have same team.slug.

Fixes #9985 

## Requirement/Documentation


  - If there is a requirement document, please, share it here.
  - If there is ab UI/UX design document, please, share it here.

## Type of change




## How should this be tested?

  - Are there environment variables that should be set? No
  - What are the minimal test data to have? A newly created orgs with a couple teams already created
  - What is expected (happy path) to have (input and output)? When inviting a user via link and he ends up on signup page, the username is validated against team slugs in the entire org or just that simple team. If username has the same name as the team he should see an error that says Username is taken.
  - Any other important info that could help to test that PR

## Mandatory Tasks


## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
